testsuite/MDAnalysisTests/coordinates/test_copying.py

```diff
@@ -0,0 +1,158 @@
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import absolute_import
+
+import numpy as np
+try:
+    from numpy import shares_memory
+except ImportError:
+    shares_memory = False
+    
+from numpy.testing import assert_equal
+import pytest
+import MDAnalysis as mda
+
+from MDAnalysisTests.datafiles import (
+    AUX_XVG,
+    CRD,
+    DCD,
+    DMS,
+    DLP_CONFIG,
+    DLP_HISTORY,
+    INPCRD,
+    GMS_ASYMOPT,
+    GRO,
+    LAMMPSdata_mini,
+    mol2_molecules,
+    MMTF,
+    NCDF,
+    PDB_small,
+    PDBQT_input,
+    PQR,
+    TRR,
+    TRJ,
+    TRZ,
+    XTC,
+    XPDB_small,
+    XYZ_mini,
+)
+from MDAnalysis.coordinates.core import get_reader_for
+
+
+@pytest.fixture(params=[
+    # formatname, filename
+    ('CRD', CRD, dict()),
+    ('DATA', LAMMPSdata_mini, dict(n_atoms=1)),
+    ('DCD', DCD, dict()),
+    ('DMS', DMS, dict()),
+    ('CONFIG', DLP_CONFIG, dict()),
+    ('HISTORY', DLP_HISTORY, dict()),
+    ('INPCRD', INPCRD, dict()),
+    ('GMS', GMS_ASYMOPT, dict()),
+    ('GRO', GRO, dict()),
+    ('MMTF', MMTF, dict()),
+    ('MOL2', mol2_molecules, dict()),
+    ('PDB', PDB_small, dict()),
+    ('PQR', PQR, dict()),
+    ('PDBQT', PDBQT_input, dict()),
+    ('TRR', TRR, dict()),
+    ('TRZ', TRZ, dict(n_atoms=8184)),
+    ('TRJ', TRJ, dict(n_atoms=252)),
+    ('XTC', XTC, dict()),
+    ('XPDB', XPDB_small, dict()),
+    ('XYZ', XYZ_mini, dict()),
+    ('NCDF', NCDF, dict()),
+    ('memory', np.arange(60).reshape(2, 10, 3).astype(np.float64), dict()),
+], scope='module')
+def refReader(request):
+    fmt_name, filename, extras = request.param
+
+    r = get_reader_for(filename, format=fmt_name)(filename, **extras)
+    try:
+        yield r
+        # make sure file handle is closed afterwards
+    finally:
+        r.close()
+
+
+@pytest.fixture()
+def original_and_copy(refReader):
+    new = refReader.copy()
+    try:
+        yield refReader, new
+    finally:
+        new.close()
+
+
+def test_reader_n_atoms(original_and_copy):
+    original, copy = original_and_copy
+    assert original.n_atoms == copy.n_atoms
+
+def test_reader_filename(original_and_copy):
+    original, copy = original_and_copy
+    assert original.filename == copy.filename
+
+def test_reader_independent_iteration(original_and_copy):
+    original, copy = original_and_copy
+    if len(original) < 2:
+        pytest.skip('Single frame reader')
+    # initially at same frame
+    assert original.ts.frame == copy.ts.frame
+
+    copy[1]
+    assert original.ts.frame == 0
+    assert copy.ts.frame == 1
```

Comment:
A test would be nice that the frame is the same without any seeking. especially if the original wasn't at frame 0.

I would also call `next` on both readers and check that they loaded the same timestep.